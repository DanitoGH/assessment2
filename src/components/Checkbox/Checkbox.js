import React from 'react';
import PropTypes from 'prop-types';

import { FormCheck, FormLabel  } from 'react-bootstrap';


const Checkbox = ({ label , roomNumber,  checked, onChange }) => {

    const handleChange = (e) => {
        const { checked } = e.target
        onChange({checked: checked, roomNumber: roomNumber})
    }

    return (
        <div>
           { (roomNumber !== 1)? 
             <FormCheck 
                type="checkbox"
                label={label}
                value={checked}
                checked={checked}
                onChange={handleChange}
            /> : <FormLabel>Room 1</FormLabel>  }

        </div>
    );
};

Checkbox.protoTypes = {
   label: PropTypes.string.isRequired,
   roomNumber: PropTypes.string.isRequired,
   onChange: PropTypes.func.isRequired
}

 export default Checkbox;