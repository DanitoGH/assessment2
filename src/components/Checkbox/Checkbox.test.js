import { render } from '@testing-library/react';
import { shallow }  from "enzyme";
import toJSON from "enzyme-to-json";

import Checkbox from './Checkbox';


test('properly renders checkbox component', () => {
  render(<Checkbox />);
})

test("matches snapshot", () => {
  const wrapper = shallow(<Checkbox />);

  expect(toJSON(wrapper)).toMatchSnapshot();
})
