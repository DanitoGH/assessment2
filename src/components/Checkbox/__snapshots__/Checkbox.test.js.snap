// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <FormCheck
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
